plutus-ir/src/Language/PlutusIR/Compiler/Types.hs

```diff
@@ -19,6 +19,6 @@ type PLCType a = PLC.Type PLC.TyName (Provenance a)
 type PIRTerm a = PIR.Term PIR.TyName PIR.Name (Provenance a)
 type PIRType a = PIR.Type PIR.TyName (Provenance a)
 
-type Compiling m a = (Monad m, MonadReader (Provenance a) m, MonadError (Error (Provenance a)) m, MonadQuote m)
+type Compiling m e a = (Monad m, MonadReader (Provenance a) m, AsError e (Provenance a), MonadError e m, MonadQuote m)
```

Comment:
Here is your unused type variable :)
It's a shame we can't turn a functional dependency into a type family and get rid of this `e`.

Reply:
I probably could instead define my own typeclass that has these constraints as super-constraints and has an associated type, but I'm not sure I want to.

Reply:
I wouldn't want as well.